src/extensions/extension-loader/extension-loader.ts

```diff
@@ -339,21 +341,32 @@ export class ExtensionLoader {
       // Remove null values
       .filter(extension => Boolean(extension));
 
-    // We first need to wait until each extension's `onActivate` is resolved,
+    // We first need to wait until each extension's `onActivate` is resolved or rejected,
     // as this might register new catalog categories. Afterwards we can safely .enable the extension.
-    await Promise.all(extensions.map(extension => extension.activated));
+    await Promise.all(
```

Comment:
Given that we have already started all the activations above, why not just use a for loop and `try...catch`, then we don't need another field, or the filter, or the map. Just an array to `push` to.

Reply:
I checked the original code, and it seems we want to return even extensions that fail to be activated from `loadExtensions`. I removed the filter and the field to simplify it.

Reply:
So looking at the code around this, we could easily make `activate` call `enable`, have the logging within `LensExtension` and then pass those promises directly to the `return extensions.map(...)` bellow, because we don't actually need this awaits here.

Reply:
No it would print the errors as they come in.

Technically, we would would `allSettled`, but anyway, I don't see why we need that there in the first place

Reply:
They wouldn't print as they come, as await inside the for-of loop waits until each iteration has been completed, it doesn't run "in parallel".

Reply:
Oh they will run concurrently because you call the `activate` function above before the loop. They just won't be reported concurrently.

Reply:
Yeah, I was talking about the code block you suggested, not earlier / other code.

Reply:
Yes, however you are touching this code. And I think that the current logic is very complex and could be simplified without harming our debugging experience.

Reply:
What do you suggest to change? Promise.all is fine for this use case, and I explained why using your try/catch is inferior.

The 'we could easily make activate call enable, have the logging within LensExtension and then pass those promises directly to the return extensions.map(...)' is a refactoring that I don't think is needed for this fix.

Reply:
And in fact having activate call enable wouldn't work because of the following reason:
```
    // We first need to wait until each extension's `onActivate` is resolved,
    // as this might register new catalog categories. Afterwards we can safely .enable the extension.
```

Reply:
I would do the following:

```ts
protected async loadExtensions(installedExtensions: Map<string, InstalledExtension>, register: (ext: LensExtension) => Promise<Disposer[]>) {
  const extensions: ExtensionLoading[] = [];

  for (const [extId, ext] of installedExtensions) {
    const alreadyInit = this.instances.has(extId) || this.nonInstancesByName.has(ext.manifest.name);

    if (ext.isCompatible && ext.isEnabled && !alreadyInit) {
      try {
        const LensExtensionClass = this.requireExtension(ext);

        if (!LensExtensionClass) {
          this.nonInstancesByName.add(ext.manifest.name);

          continue;
        }

        const instance = this.dependencies.createExtensionInstance(
          LensExtensionClass,
          ext,
        );

        this.instances.set(extId, instance);

        extensions.push({
          isBundled: ext.isBundled,
          loaded: (async (): Promise<void> => {
            try {
              await instance.activate();
            } catch (error) {
              return void logger.error(`${logModule}: extension activation error`, { ext, error });
            }

            try {
              await instance.enable(register);
            } catch (error) {
              return void logger.error(`${logModule}: extension enabling error`, { ext, error });
            }
          })(),
        });
      } catch (error) {
        logger.error(`${logModule}: extension loading error`, { ext, error });
      }
    } else if (!ext.isEnabled && alreadyInit) {
      this.removeInstance(extId);
    }
  }

  return extensions;
}
```

Reply:
That doesn't work for this reason:
```
    // We first need to wait until each extension's `onActivate` is resolved,
    // as this might register new catalog categories. Afterwards we can safely .enable the extension.
```

Your code example might call .enable before another extension is activated fully, thus causing error.

See earlier PR for some discussion:

Reply:
I propose we fix that in another PR as that's becoming not related to the issue at hand.

Reply:
Okay sounds good